### PR TITLE
fix: default to custom for empty transaction and span types

### PR DESCRIPTION
**Pull request by kruskall**
According to the spec:

If no transaction.type or span.type is provided or the value
is an empty string, the agent needs to set a default value
custom.

Add test to assert the behaviour is compliant to the spec.

Closes https://github.com/elastic/apm-agent-go/issues/1211

### CODE REVIEW

File: span_test.go [hunk 1]
```diff
@@ -118,6 +118,22 @@ func TestSpanParentID(t *testing.T) {
...
+	tx := tracer.StartTransaction("name", "type")
+	span := tx.StartSpan("name", "", nil)
+	span.End()
+	tx.End()
+	tracer.Flush(nil)
+
+	payloads := tracer.Payloads()
+	require.Len(t, payloads.Spans, 1)
+
+	assert.NotEmpty(t, payloads.Spans[0].Type)
```
Comment (axw) [inline span_test.go:134]:
```suggestion
	assert.Equal(t, "custom", payloads.Spans[0].Type)
```

File: transaction_test.go [hunk 2]
```diff
@@ -143,6 +143,19 @@ func TestTransactionEnsureParent(t *testing.T) {
...
+	tracer, transport := transporttest.NewRecorderTracer()
+	defer tracer.Close()
+
+	tx := tracer.StartTransaction("name", "")
+	tx.End()
+	tracer.Flush(nil)
+
+	payloads := transport.Payloads()
+	require.Len(t, payloads.Transactions, 1)
+	assert.NotEmpty(t, payloads.Transactions[0].Type)
```
Comment (axw) [inline transaction_test.go:156]:
```suggestion
	assert.Equal(t, "custom", payloads.Transactions[0].Type)
```

**Review by axw — COMMENTED** [review]
Looks good, but the `module/apmot` tests are failing, as they are depending on Type to be empty after construction.

Maybe there's a better alternative, but we I think we may need to set this default value when ending the transaction or span, if the value is empty. That's a bit more foolproof anyway, since users could directly set `Type = ""` after starting the transaction or span.

File: span_test.go (hunk 1 above)
Comment (kruskall) [reply span_test.go]: Changed!

File: transaction_test.go (hunk 2 above)
Comment (kruskall) [reply transaction_test.go]: Fixed!

**Review by axw — APPROVED** [review]
LGTM!

**Comment by ugosan** [comment]
this should also fix this one https://github.com/elastic/kibana/issues/126805

**Pull request merged**